atc/api/accessor/accessor_test.go

```diff
@@ -1,900 +1,649 @@
-			BeforeEach(func() {
-				claims = &jwt.MapClaims{"csrf": ""}
-			})
 			It("returns false", func() {
-				Expect(access.CSRFToken()).To(BeEmpty())
-			})
-		})
-		Context("when request has csrfToken claim set to nil", func() {
-			BeforeEach(func() {
-				claims = &jwt.MapClaims{"csrf": nil}
-			})
-			It("returns false", func() {
-				Expect(access.CSRFToken()).To(BeEmpty())
+				Expect(result).To(BeFalse())
 			})
 		})
 
-		Context("when request does not have csrfToken claim set", func() {
+		Context("when the user has invalid token", func() {
 			BeforeEach(func() {
-				claims = &jwt.MapClaims{}
+				verification.HasToken = true
+				verification.IsTokenValid = false
 			})
+
 			It("returns false", func() {
-				Expect(access.CSRFToken()).To(BeEmpty())
+				Expect(result).To(BeFalse())
+			})
+		})
+
+		Context("when the user has valid token", func() {
+			BeforeEach(func() {
+				verification.HasToken = true
+				verification.IsTokenValid = true
+				verification.Claims = map[string]interface{}{
+					"federated_claims": map[string]interface{}{
+						"connector_id": "some-connector",
+						"user_id":      "some-user-id",
+					},
+				}
+			})
+
+			Context("when the user is a not on an admin team", func() {
+				BeforeEach(func() {
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"viewer": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+					fakeTeam2.AuthReturns(atc.TeamAuth{
+						"member": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+					fakeTeam3.AuthReturns(atc.TeamAuth{
+						"owner": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+				})
+
+				It("returns false", func() {
+					Expect(result).To(BeFalse())
+				})
+			})
+
+			Context("when the user is a 'viewer' on an admin team", func() {
+				BeforeEach(func() {
+					fakeTeam1.AdminReturns(true)
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"viewer": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+				})
+
+				It("returns false", func() {
+					Expect(result).To(BeFalse())
+				})
+			})
+
+			Context("when the user is a 'member' on an admin team", func() {
+				BeforeEach(func() {
+					fakeTeam1.AdminReturns(true)
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"member": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+				})
+
+				It("returns false", func() {
+					Expect(result).To(BeFalse())
+				})
+			})
+
+			Context("when the user is a 'owner' on an admin team", func() {
+				BeforeEach(func() {
+					fakeTeam1.AdminReturns(true)
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"owner": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+				})
+
+				It("returns true", func() {
+					Expect(result).To(BeTrue())
+				})
```

Comment:
Should `pipeline-operator` role be tested when on an admin team also?

Reply:
done